app/src/main/java/tech/relaycorp/gateway/domain/endpoint/NotifyEndpoint.kt

```diff
@@ -1,11 +1,28 @@
 package tech.relaycorp.gateway.domain.endpoint
 
+import android.content.Context
+import android.content.Intent
+import tech.relaycorp.gateway.background.RelaynetAndroid
+import tech.relaycorp.gateway.common.Logging.logger
+import tech.relaycorp.gateway.data.database.LocalEndpointDao
 import tech.relaycorp.gateway.data.model.MessageAddress
 import javax.inject.Inject
 
 class NotifyEndpoint
-@Inject constructor() {
+@Inject constructor(
+    private val endpointDao: LocalEndpointDao,
+    private val context: Context
+) {
 
-    // TODO: implementation
-    suspend fun notify(endpointAddress: MessageAddress) {}
+    suspend fun notify(endpointAddress: MessageAddress) {
+        endpointDao.get(endpointAddress)
+            ?.let { endpoint ->
+                context.sendBroadcast(
+                    Intent(RelaynetAndroid.ENDPOINT_NOTIFY_ACTION)
+                        .setPackage(endpoint.applicationId),
+                    RelaynetAndroid.GATEWAY_SYNC_PERMISSION
+                )
```

Comment:
Just wanted to double check: This isn't broadcasting the event to many/all apps on the phone -- Instead, it's limiting the message to the endpoint's app. Right?

I think that's what happening but I wanted to check because of the word "broadcast".

Or is there any chance that third party developers can register intents for other apps? Or simply monitor broadcasts for other apps?

Reply:
Any app can subscribe to a certain broadcast, but since we're using the `setPackage` filter, only components inside that application will receive them.